Src/BootCamp.Chapter/LinqMethodDemo.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    public static class LinqMethodDemo
+    {
+        internal static IEnumerable<T> Demo<T>(this IEnumerable<T> list, Predicate<T> predicate)
+        {
+            list = Filter(list, predicate);
+            list = ReduceToHalfLength(list);
+
+            return list; 
+        }
+
+        private static List<T> ReduceToHalfLength<T>(IEnumerable<T> list)
```

Comment:
``RemoveHalfElementsAtRandom`` is a more accurate name.

Reply:
oke, changed but right now I do not delete atRandom. I delete all items after the half of the collection

Reply:
It should be random as asked in the task.

Reply:
Then I overread it several times. 
I do not see anything about random here : 
``` 
 Make your own LINQ method! Create a method SnapFingers which works like LINQ methods, 
takes a predicate and removes exactly half of all the elements in the collection.
If it's not even number, it should remove 1 less.  
For example:  
if it's 2, it removes 1;  
if it's 4, it removes 2;  
but if it's 3, it removes 1
```